lib/puppet/defaults.rb

```diff
@@ -1967,6 +1967,12 @@ def self.initialize_default_settings!(settings)
       already running puppet agent to finish before starting a new one. This is set by default to 1 minute.
       A value of `unlimited` will cause puppet agent to wait indefinitely. 
       #{AS_DURATION}",
+    },
+    :default_agent_environment => {
+      :default  => "production",
+      :type     => :string,
+      :desc     => "The default environment for puppet agent. Set by default to 'production', and
+      is the fallback environment in case the agent runs in an environment not available on the server.",
```

Comment:
The sentence "The default environment for puppet agent" makes it sound like it's an agent setting. Maybe combine the two, "The default environment to assign to agent requests if the requested environment is not available on the server."?

Also since this is a server side setting only, we should define it in the `server` section https://github.com/puppetlabs/puppet/blob/728586c867bfe2e0a1085fc333cada662b91d449/lib/puppet/defaults.rb#L1344

Maybe near the `user`, `group`, `default_manifest`, etc settings?

Note the `define_section(:server)` mostly provides a logical grouping of settings. It doesn't affect how settings are resolved based on run_mode, etc, e.g. `Puppet[:default_agent_enviroment]`. The only real impact is for `:file` and `:directory` related settings when someone calls `Puppet.settings.use(:main, ...)`, since in that case we apply a settings catalog for all file/directory settings in the `:main` section, for example, to set the owner/group/mode for `Puppet[:ssldir]`.